README.adoc

```diff
@@ -0,0 +1,402 @@
+    this.height = height
+    this.width = width
+  }
+}
+```
+
+**Pass**
+
+```js
+const polygon = (height, width) => ({
+  height: height,
+  width: width,
+})
+```
+
+### `fp/no-delete`
+
+> delete is an operator to remove fields from an object or elements from an array. This purposely mutates data, which is not wanted when doing functional programming.
+
+Further reading: https://github.com/google/google-api-nodejs-client/issues/375[Avoid using delete operator]
+
+**Fail**
+```js
+delete foo
+delete foo.bar
+delete foo[bar]
+```
+**Pass**
+```js
+import F from "ramda"
+
+const fooWithoutBar = F.omit(["bar"], foo)
+const fooWithoutField = F.omit([bar], foo)
+```
+
+### `fp/no-events`
+
+> The use of EventEmitter with the events module provided by Node.js promotes implicit side-effects by emitting and listening to events. Instead of events, you should prefer activating the wanted effects by calling the functions you wish to use explicitly.
+
+Probably what you should do is use a https://gist.github.com/staltz/868e7e9bc2a7b8c1f754[functional reactive programming] library: https://github.com/cujojs/most[`most`], https://github.com/Reactive-Extensions/RxJS[`rxjs`].
+
+**Fail**
+
+```js
+import EventEmitter from "events"
+```
+
+### `fp/no-get-set`
+
+**Fail**
+```js
+const person = {
+  name: 'Some Name',
+  get age () {
+    return this._age
+  },
+  set age (n) {
+    if (n < 0) {
+      this._age = 0
+    } else if (n > 100) {
+      this._age = 100
+    } else {
+      this._age = n
+    }
+  }: 20
+};
+
+person.__defineGetter__("name", function () {
+  return this.name || "John Doe";
+})
+
+person.__defineSetter__("name", function (name) {
+  this.name = name.trim();
+})
+```
+**Pass**
+```js
+import F from "ramda"
+
+const person = {
+  name: "Some Name",
+  age: 20,
+}
+
+const clamp = (n, min, max) =>
+  n <= min ? min :
+  n >= max ? max :
+             n
+
+const setAge = (age, person) =>
+  F.merge(person, { age: clamp(age, 0, 100) })
+```
+
+### `fp/no-let`
+
+> If you want to program as if your variables are immutable, part of the answer is to not allow your variables to be reassigned. By not allowing the use of let and var, variables that you declared may not be reassigned.
+
+**Fail**
+```js
+let a = 1
+let b = 2,
+    c = 3
+let d
+```
+**Pass**
+```js
+const a = 1
+const b = 2,
+      c = 3
+```
+
+### `fp/no-loops`
+> Loops, such as for or while loops, work well when using a procedural paradigm. In functional programming, recursion or implementation agnostic operations like map, filter and reduce are preferred.
+
+**Fail**
+```js
+const result = []
+const elements = [1, 2, 3]
+
+for (let i = 0; i < elements.length; i++) {
+  if (elements[i] > 2) {
+    result.push(elements[i])
+  }
+}
+
+for (element in elements) {
+  result.push(element * 10)
+}
+
+while (n < 100) {
+  result.push(n)
+  n *= 2
+}
+```
+**Pass**
+```js
+const xs = [1, 2, 3]
+
+xs.filter(x => x > 2)
+
+xs.map(x => x * 10)
+
+const doubleBlast (n) =>
+  n >= 100
```

Comment:
> the JS way

Given JS is a multi-paradigm language informed with only 10 days of initial design, and has _necessary_ popularity for historical reasons (browser monopoly), there is no `JS Way` like you might find in Ruby or Python. Instead, the best case is consistent org/community-level standards. That's what a highly modular linter exists in JavaScript, whereas other languages have e.g. [`go fmt`](https://blog.golang.org/go-fmt-your-code) that everyone shares. In other words JS standards between orgs (FB, MS, Goog, ...) and communities (Angular, React, Ember, Cycle...) can be highly variant.

If we think critically, avoid [cargo culting](https://en.wikipedia.org/wiki/Cargo_cult_programming), and seek overall consensus as a team/org/community, I'm happy.

> also can we add an eslint rule where we must use curly braces for if statements.

Amazing example, ha! About the rule, I think I agree generally. But function-body top-level `if`s used as guards are clearer as single line I think (this is an idiom in the core node community, another example of lack of one "JS Way"):
```js
foo (a, b, c) {
  if (/*...*/) return // ...
  if (/*...*/) return // ...
  if (/*...*/) return // ...
  /*...*/
  return //...
}
```

But this doesn't have to be mutually exclusive meaning I think this use-case could be supported specially via a new rule we create (e.g. `if-guards` that permit single-line `if` blocks starting with `return` located within function-body top-level).

Anyways, interesting conversation! I don't think we have any major differences? This has veered off-topic of the rule in question (`no-loops`) so lets keep any more general opinions on the main PR thread if possible.